tests/ci_tests/code_formatter/crewai/test_crewai_agents.py

```diff
@@ -0,0 +1,310 @@
+
+        agent_map = agents.create_code_formatter_agents(mock_langfuse_client)
+        agent = agent_map["senior_engineer"]
+
+        # Verify the prompts were fetched
+        assert mock_langfuse_client.get_prompt.call_count == 2
+        mock_langfuse_client.get_prompt.assert_any_call("refactor_agent/goal_prompt")
+        mock_langfuse_client.get_prompt.assert_any_call("refactor_agent/backstory_prompt")
+
+        # Verify the agent uses the prompts
+        assert agent.goal == expected_goal
+        assert agent.backstory == expected_backstory
+
+    def test_agent_has_no_tools(self):
+        """Test that the senior engineer agent has no tools"""
+        mock_langfuse_client = mock.MagicMock()
+
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = "goal"
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = "backstory"
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        agent_map = agents.create_code_formatter_agents(mock_langfuse_client)
+        agent = agent_map["senior_engineer"]
+
+        assert agent.tools == []
+
+    def test_agent_does_not_delegate(self):
+        """Test that the agent has delegation disabled"""
+        mock_langfuse_client = mock.MagicMock()
+
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = "goal"
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = "backstory"
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        agent_map = agents.create_code_formatter_agents(mock_langfuse_client)
+        agent = agent_map["senior_engineer"]
+
+        assert agent.allow_delegation is False
+
+    def test_agent_is_verbose(self):
+        """Test that the agent has verbose mode enabled"""
+        mock_langfuse_client = mock.MagicMock()
+
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = "goal"
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = "backstory"
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        agent_map = agents.create_code_formatter_agents(mock_langfuse_client)
+        agent = agent_map["senior_engineer"]
+
+        assert agent.verbose is True
+
+    def test_langfuse_error_handling(self):
+        """Test that Langfuse errors are propagated"""
+        mock_langfuse_client = mock.MagicMock()
+        mock_langfuse_client.get_prompt.side_effect = Exception("Cannot fetch prompts from Langfuse")
+
+        with pytest.raises(Exception, match="Cannot fetch prompts from Langfuse"):
```

Comment:
Added a docstring to the method, describing its purpose.
```suggestion
    def test_langfuse_error_handling(self):
```